### ISSUE TITLE
How to get stated with NNUE

### ISSUE THREAD

**Issue by ArjunBasandrai** [jhonnold/berserk#537]
Hi Jay!

I am currently working on the Shuffle Chess Engine. I've made the decision to incorporate NNUE into my engine, but I'm facing challenges finding resources to kick-start the process. I'm particularly interested in understanding the required position encoding for NNUE and how to structure the architecture. It seems that the neural net differs from conventional networks. Additionally, the concept of horizontally mirrored buckets is a bit unclear to me.

Could you please share some resources to help me get started? Are there any talkchess threads or other materials that you think would be beneficial for my understanding? Your guidance would be greatly appreciated.

Thank you!

**Comment by jhonnold** [comment]
- Some documentation: https://github.com/official-stockfish/nnue-pytorch/blob/master/docs/nnue.md
- SF Discord Link: https://discord.gg/GWDRS3kU6R
  - Go here to see their `#resources` channel
- 3B1B on NNs: https://www.youtube.com/playlist?list=PLZHQObOWTQDNU6R1_67000Dx_ZCJB-3pi

*This really isn't the best place to ask for advice - there are many discord communities that are floating around where looking for help is better*
- Engine Programming: https://discord.com/invite/F6W6mMsTGN
- See above for SF discord
- OpenBench discord: https://discord.gg/9MVg7fBTpM
